glyphs-reader/src/font.rs

```diff
@@ -0,0 +1,267 @@
+//! The general strategy is just to use a plist for storage. Also, lots of
+//! unwrapping.
+//!
+//! There are lots of other ways this could go, including something serde-like
+//! where it gets serialized to more Rust-native structures, proc macros, etc.
+
+use std::collections::BTreeMap;
+use std::hash::Hash;
+
+use ordered_float::OrderedFloat;
+
+use crate::from_plist::FromPlist;
+use crate::plist::Plist;
+use crate::to_plist::ToPlist;
+
+#[derive(Debug, FromPlist, ToPlist, Hash)]
+pub struct Font {
+    pub glyphs: Vec<Glyph>,
+    pub font_master: Vec<FontMaster>,
+    #[rest]
+    pub other_stuff: BTreeMap<String, Plist>,
+}
+
+#[derive(Clone, Debug, FromPlist, ToPlist, Hash)]
+pub struct Glyph {
+    pub layers: Vec<Layer>,
+    pub glyphname: String,
+    #[rest]
+    pub other_stuff: BTreeMap<String, Plist>,
+}
+
+#[derive(Clone, Debug, FromPlist, ToPlist, Hash)]
+pub struct Layer {
+    pub layer_id: String,
+    pub width: OrderedFloat<f64>,
+    pub paths: Option<Vec<Path>>,
+    pub components: Option<Vec<Component>>,
+    pub anchors: Option<Vec<Anchor>>,
+    pub guide_lines: Option<Vec<GuideLine>>,
```

Comment:
the two-word camel case seems odd, I'd expect one `guidelines`. Was it auto-snake_cased?

BTW, I think in Glyphs3 source format, these are now just called `guides`

see https://github.com/schriftgestalt/GlyphsSDK/blob/Glyphs3/GlyphsFileFormat/GlyphsFileFormatv3.md#differences-between-version-2

> guideLines has been renamed guides.

Reply:
usually we don't really care aboud guidelines for compilation purposes, you may even put them in the `#[rest]` of the other_stuff

Reply:
Good point, I think I will just drop them

Reply:
Fixed